packages/backend/server/src/plugins/copilot/controller.ts

```diff
@@ -245,15 +196,16 @@ export class CopilotController implements BeforeApplicationShutdown {
     @CurrentUser() user: CurrentUser,
     @Req() req: Request,
     @Param('sessionId') sessionId: string,
-    @Query() params: Record<string, string | string[]>
+    @Query() query: Record<string, string | string[]>
   ): Promise<string> {
-    const info: any = { sessionId, params };
+    const info: any = { sessionId, params: query };
 
     try {
-      const { messageId, retry, reasoning, webSearch, modelId } =
-        this.prepareParams(params);
+      let { messageId, retry, reasoning, webSearch, modelId, params } =
+        ChatQuerySchema.parse(query);
 
-      const { provider, model } = await this.chooseTextProvider(
+      const { provider, model } = await this.chooseProvider(
+        ModelOutputType.Text,
```

Comment:
There should be two parameters here, `inputType: ModelInputType[]` and `outputType: ModelOutputType`?

Reply:
done